pkg/controller/master/backup/backup_target.go

```diff
@@ -250,14 +250,30 @@ func (h *TargetHandler) updateBackupTargetSecret(target *settings.BackupTarget)
 	return h.updateLonghornBackupTargetSecretSetting(target)
 }
 
-func (h *TargetHandler) deleteBackupTargetSecret(target *settings.BackupTarget) error {
-	if err := h.secrets.Delete(util.LonghornSystemNamespaceName, util.BackupTargetSecretName, nil); err != nil && !apierrors.IsNotFound(err) {
+func (h *TargetHandler) resetBackupTargetSecret() error {
+	secret, err := h.secretCache.Get(util.LonghornSystemNamespaceName, util.BackupTargetSecretName)
+	if err != nil && !apierrors.IsNotFound(err) {
 		return err
 	}
+	if secret != nil {
+		secretCpy := secret.DeepCopy()
+		secretCpy.Data = map[string][]byte{}
```

Comment:
please double check, should we check the .Data len == 0 to avoid unnecessary update ? following update is similar

thanks.

Reply:
Yes, good catch. Added checking whether `secret.Data` and `setting.Value` is empty before udpate.